### PR TITLE
Added security advisory for gateway port change

### PR DESCRIPTION
**Pull request by nrjpoddar**
Added a script that users can run to evaluate if any of their existing
Authorization Policies need to be migrated.

[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by nrjpoddar** [comment]
@yangminzhu @howardjohn PTAL.

File: content/en/news/security/istio-security-2021-002/index.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+releases: ["All releases 1.6 and later"]
+publishdate: 2021-04-06
+keywords: [CVE]
+skip_seealso: true
+---
+
+{{< security_bulletin >}}
+
+Upgrading to Istio version 1.6 and higher is vulnerable to the following
+misconfiguration related vulnerability:
```
Comment (howardjohn) [inline content/en/news/security/istio-security-2021-002/index.md:17]:
```suggestion
Upgrading to Istio version 1.6 and may result in misconfigured Authorization Polices leading to unexpected behavior:
```
Just my 2c, "vulnerability" sounds wrong here

File: content/en/news/security/istio-security-2021-002/index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+policies, update the copied version to use new gateway workload ports and
+apply both existing and updated policies in your cluster before initiating
+the upgrade process. You should only delete the old policies after a
+successful upgrade to ensure no policy violations occur on upgrade
+failures or rollbacks.
+
+## Credit
+
+We'd like to thank [Neeraj Poddar](https://twitter.com/nrjpoddar)
+for reporting this vulnerability.
```
Comment (howardjohn) [inline content/en/news/security/istio-security-2021-002/index.md:101]: I again disagree with the use of "vulnerability" here.

**Review by howardjohn — APPROVED** [review]

File: content/en/news/security/istio-security-2021-002/index.md (hunk 1 above)
Comment (nrjpoddar) [reply content/en/news/security/istio-security-2021-002/index.md:17]: agree

File: content/en/news/security/istio-security-2021-002/index.md (hunk 2 above)
Comment (nrjpoddar) [reply content/en/news/security/istio-security-2021-002/index.md:101]: will change

**Review by linsun — APPROVED** [review]
Thanks Neeraj!

**Comment by nrjpoddar** [comment]
Any doc maintainers should can approve this?

@ericvn ?

**Comment by nrjpoddar** [comment]
I’m assuming we need to cherry-pick this to 1.9?

File: content/en/news/security/istio-security-2021-002/index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+---
+title: ISTIO-SECURITY-2021-002
+subtitle: Security Bulletin
+description: Ingress gateway authorization policy violation on upgrades.
+cves: [N/A]
+cvss: "N/A"
+vector: ""
+releases: ["All releases 1.6 and later"]
+publishdate: 2021-04-06
```
Comment (craigbox) [inline content/en/news/security/istio-security-2021-002/index.md:9]:
```suggestion
publishdate: 2021-04-07
```

File: content/en/news/security/istio-security-2021-002/index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+keywords: [CVE]
+skip_seealso: true
+---
+
+{{< security_bulletin >}}
+
+Upgrading to Istio version 1.6 and higher may result in misconfigured
+Authorization Policies:
```
Comment (craigbox) [inline content/en/news/security/istio-security-2021-002/index.md:17]:
```suggestion
Upgrading from Istio versions 1.5 and prior, to 1.6 and later, may result in misconfigured access control:
```

File: content/en/news/security/istio-security-2021-002/index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+gateways are updated from port "80" to "8080" and "443" to "8443" to allow
+[gateways to run as non-root](/news/releases/1.7.x/announcing-1.7/upgrade-notes/#gateways-run-as-non-root)
+by default. With this change, any existing Authorization Policies targeting
+Istio ingress gateway on ports "80" and "443" need to be migrated to use the
+new container ports "8080" and "8443" before upgrading to the listed versions.
+Failure to migrate might result in traffic reaching at ingress gateway service
+ports "80" and "443" to be incorrectly allowed or blocked thereby causing policy
+violations.
+
+Example of an Authorization Policy resource that needs to be updated:
```
Comment (craigbox) [inline content/en/news/security/istio-security-2021-002/index.md:30]:
```suggestion
- **Incorrect gateway ports on authorization policies on upgrades**: In Istio
versions 1.6 and later, the default container ports for Istio ingress
gateways are updated from port "80" to "8080" and "443" to "8443" to allow
[gateways to run as non-root](/news/releases/1.7.x/announcing-1.7/upgrade-notes/#gateways-run-as-non-root)
by default. With this change, any existing authorization policies targeting
an Istio ingress gateway on ports `80` and `443` need to be migrated to use the
new container ports `8080` and `8443`, before upgrading to the listed versions.
Failure to migrate may result in traffic reaching ingress gateway service
ports `80` and `443` to be incorrectly allowed or blocked, thereby causing policy
violations.

Example of an authorization policy resource that needs to be updated:
```

File: content/en/news/security/istio-security-2021-002/index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        - operation:
+            paths: ["/admin"]
+            ports: [ "443" ]
+
+    {{< /text >}}
+
+The above policy in Istio versions 1.5 and prior will block all access to path
+"/admin" for traffic reaching Istio ingress gateway on container ports "80"
+and "443". On upgrading to Istio version 1.6 and later, this policy should
+be updated to the following to have the same effect.
```
Comment (craigbox) [inline content/en/news/security/istio-security-2021-002/index.md:58]:
```suggestion
The above policy in Istio versions 1.5 and prior will block all access to path
`/admin` for traffic reaching an Istio ingress gateway on container ports `80`
and `443`. On upgrading to Istio version 1.6 and later, this policy should
be updated to the following to have the same effect:
```

File: content/en/news/security/istio-security-2021-002/index.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+attached to the default Istio ingress gateway in istio-system namespace needs
+to be updated. If you’re using a custom gateway installation you can customize
+the script to run with parameters applicable to your environment.
+
+It is recommended to create a copy of your existing misconfigured Authorization
+policies, update the copied version to use new gateway workload ports and
+apply both existing and updated policies in your cluster before initiating
+the upgrade process. You should only delete the old policies after a
+successful upgrade to ensure no policy violations occur on upgrade
+failures or rollbacks.
```
Comment (craigbox) [inline content/en/news/security/istio-security-2021-002/index.md:96]:
```suggestion
- Update your authorization policies before upgrading to the
affected Istio versions. You can use this [script](./check.sh)
to check if any of the existing authorization policies
attached to the default Istio ingress gateway in the `istio-system` namespace need
to be updated. If you’re using a custom gateway installation, you can customize
the script to run with parameters applicable to your environment.

It is recommended to create a copy of your existing authorization
policies, update the copied version to use new gateway workload ports, and
apply both existing and updated policies in your cluster, before initiating
the upgrade process. You should only delete the old policies after a
successful upgrade, to ensure no policy violations occur on upgrade
failures or rollbacks.
```

File: content/en/news/security/istio-security-2021-002/index.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
+---
+title: ISTIO-SECURITY-2021-002
+subtitle: Security Bulletin
+description: Ingress gateway authorization policy violation on upgrades.
```
Comment (craigbox) [inline content/en/news/security/istio-security-2021-002/index.md:4]:
```suggestion
description: Upgrades from older Istio versions can affect access control to an ingress gateway due to a change of container ports.
```

**Review by craigbox — COMMENTED** [review]
Some formatting changes, standard capitalization, and changed "misconfigured" language as it wasn't a user configuration error, but rather a change brought on by the new version of Istio.

File: content/en/news/security/istio-security-2021-002/index.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+vector: ""
+releases: ["All releases 1.6 and later"]
+publishdate: 2021-04-07
+keywords: [CVE]
+skip_seealso: true
+---
+
+{{< security_bulletin >}}
+
+Upgrading from Istio versions 1.5 and prior, to 1.6 and later, may result in misconfigured access control:
```
Comment (nrjpoddar) [inline content/en/news/security/istio-security-2021-002/index.md:16]: @craigbox how about this:

Upgrading from Istio versions 1.5 and prior, to 1.6 and later, may result in access control bypass:

Comment (craigbox) [reply content/en/news/security/istio-security-2021-002/index.md:16]: wfm!

**Review by craigbox — APPROVED** [review]

**Pull request merged**